frb_codegen/src/generator/dart/ty.rs

```diff
@@ -48,17 +49,57 @@ pub enum TypeDartGenerator<'a> {
 }
 
 impl<'a> TypeDartGenerator<'a> {
-    pub fn new(ty: IrType, ir_file: &'a IrFile) -> Self {
+    pub fn new(ty: IrType, ir_file: &'a IrFile, dart_api_class_name: Option<String>) -> Self {
         let context = TypeGeneratorContext { ir_file };
         match ty {
-            Primitive(ir) => TypePrimitiveGenerator { ir, context }.into(),
-            Delegate(ir) => TypeDelegateGenerator { ir, context }.into(),
-            PrimitiveList(ir) => TypePrimitiveListGenerator { ir, context }.into(),
-            Optional(ir) => TypeOptionalGenerator { ir, context }.into(),
-            GeneralList(ir) => TypeGeneralListGenerator { ir, context }.into(),
-            StructRef(ir) => TypeStructRefGenerator { ir, context }.into(),
-            Boxed(ir) => TypeBoxedGenerator { ir, context }.into(),
-            EnumRef(ir) => TypeEnumRefGenerator { ir, context }.into(),
+            Primitive(ir) => TypePrimitiveGenerator {
+                ir,
```

Comment:
I realized a weird thing... This PR changes something like 

```rust
PrimitiveList(ir) => TypePrimitiveListGenerator { ir, context }.into(),
```

into

```rust
            PrimitiveList(ir) => TypePrimitiveListGenerator {
                ir,
                context,
                dart_api_class_name,
            }
            .into(),
```

I guess it is because some problem when merging from upstream? Because in master branch it is in the short form.

Reply:
Isn't it because it gets too long and it breaks into lines?

Reply:
Well, it may look neater to put dart_api_class_name as a field of `TypeGeneratorContext` I guess

Reply:
I see. So maybe not because merging from upstream problem, but because this PR really changes it. So maybe we can put dart_api_class_name as a field of TypeGeneratorContext